utils/html-sanitizer.js

```
@@ -0,0 +1,375 @@
+ * - Hidden CSS text (display:none, visibility:hidden, opacity:0)
+ * - Zero-size text (font-size:0, height:0, width:0)
+ * - Off-screen positioning (negative margins, absolute positioning)
+ * - HTML comments containing instructions
+ * - Script/style tag content
+ * - Invisible Unicode characters
+ * - ARIA-hidden content
+ * - Overflow hidden tricks
+ */
+
+// Invisible Unicode characters that could hide text
+const INVISIBLE_CHARS_REGEX = /[\u200B-\u200D\u2060\u2061-\u2064\u206A-\u206F\uFEFF\u00AD\u034F\u061C\u180E\u2028\u2029\u202A-\u202E]/g;
```

Biome (2.1.2)

[error] A character class cannot match a character and a combining character.

A character and a combining character forms a new character. Replace the character class with an alternation.

(lint/suspicious/noMisleadingCharacterClass)